App.tsx

```diff
@@ -0,0 +1,194 @@
+import React from 'react';
+import {
+  View,
+  Text,
+  TouchableOpacity,
+  Platform,
+  StyleSheet
+} from 'react-native';
+import NfcManager, {NfcTech} from 'react-native-nfc-manager';
+import {Buffer} from 'buffer';
+
+interface State {
+  apduResponse: string;
+  testOutput: boolean;
+  detectedTechnology: string;
+  nfcUID: string;
+  showInstructions: boolean;
+}
+
+export class App extends React.Component<any, State> {
+
+  constructor(props: any) {
+    super(props);
+    this.state = {
+      apduResponse: "Please run the test",
+      testOutput: false,
+      detectedTechnology: "",
+      nfcUID: "",
+      showInstructions: false
+    }
+  }
+
+  componentDidMount() {
+    NfcManager.start();
+  }
+
+  componentWillUnmount() {
+    this.cleanUp();
+  }
+
+  private cleanUp = () => {
+    NfcManager.cancelTechnologyRequest().catch(() => 0);
+    this.setState({
+      apduResponse: "Please run the test",
+      testOutput: false,
+      detectedTechnology: "",
+      nfcUID: "",
+      showInstructions: false
+    });
+  }
+
+  private test = async () => {
+    try {
+      let tech = NfcTech.IsoDep;
+
+      this.setState({showInstructions: true});
+
+      let resp = await NfcManager.requestTechnology(tech, {
+        alertMessage: 'Ready to send some APDU'
+      });
+
+      // the NFC uid can be found in tag.id
+      let tag = await NfcManager.getTag();
+
+
+      this.setState({
+        detectedTechnology: resp,
+        nfcUID: tag.id
+      });
+
+      // Test Case 7816_A_1
+      // Send the following SelectApplication APDU to the e-Passport.
+      // ‘00 A4 04 0C 07 A0 00 00 02 47 10 01'
+      // According to the ICAO recommendation, the P2 denotes "return no file
+      // information", and there is no Le byte present. Therefore, the response
+      // data field MUST be empty. The e-Passport MUST return status bytes 
+      //‘90 00’.
+
+      let apduAnswer = undefined;
+      if (Platform.OS === 'ios') {
```

Comment:
You could have take advantage of using `Platform`

```typescript
const command = [
        0x00,
        0xa4,
        0x04,
        0x0c,
        0x07,
        0xa0,
        0x00,
        0x00,
        0x02,
        0x47,
        0x10,
        0x01
      ];
      const apduAnswer = await Platform.select({
        ios: async () => {
          const { response } = await NfcManager.sendCommandAPDUIOS(command);
          return response;
        },
        default: async () => {
          const result = await NfcManager.transceive(command);
          return result;
        }
      })();
```

Reply:
Good solution, thank you.